lib/private/Files/Node/Folder.php

```diff
@@ -229,6 +229,11 @@ public function searchByTag($tag, $userId) {
 	 * @return \OC\Files\Node\Node[]
 	 */
 	private function searchCommon($method, $args) {
+		$limitToHome = ($method === 'searchQuery')? $args[0]->limitToHome(): false;
```

Comment:
```suggestion
		$limitToHome = ($method === 'searchQuery' && $args[0] instanceof SearchQuery) ? $this->limitSearchToHomeStorage($args[0]->getSearchOperation()) : false;
```

Probably the instanceof is  superfluous. No idea what $args could be ;)

Reply:
Nope ;)